client/src/components/context/isAuthContext.js

```diff
@@ -0,0 +1,5 @@
+import { createContext } from 'react';
+
+const IsAuthContext = createContext(false);
+
+export default IsAuthContext;
```

Comment:
Hello Alaa, 
I have some suggestion to refactor the context code :-
```js
import React, { useContext, useState, createContext } from 'react';
import { element, arrayOf } from 'prop-types';

const IsAuthContext = createContext(false);

export const AuthProvider = ({ children }) => {
  const authState = useState(false);
  return (
    <IsAuthContext.Provider value={authState}>
      {children}
    </IsAuthContext.Provider>
  );
};

AuthProvider.propTypes = {
  children: arrayOf(element).isRequired,
};
export const useAuth = () => useContext(IsAuthContext);

```
By this, You will not need to import useContext or isAuthContext when you want to use it.
Just import useAuth and invoke it and you will get the full state (value , function)